libsolidity/formal/SMTLib2Interface.h

```diff
@@ -68,6 +68,14 @@ class SMTLib2Interface: public SolverInterface, public boost::noncopyable
 
 	ReadCallback::Callback m_queryCallback;
 	std::vector<std::string> m_accumulatedOutput;
+
+	enum class SMTVariableType {
+		Function,
```

Comment:
Hm, I think we also need the domain and codomain of that function, otherwise it does not make too much sense.